src/instance.ts

```diff
@@ -192,6 +238,249 @@ class Instance extends common.GrpcServiceObject {
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backupExpiryDate = new PreciseDate(Date.now() + 1000 * 60 * 60 * 24)
+   * const backup = instance.backup('my-backup', 'projects/my-project/instances/my-instance/databases/my-database', backupExpiryDate);
+   */
+  backup(
+    backupId: string,
+    databasePath: string,
+    expireTime: PreciseDate
+  ): Backup {
+    if (!backupId) {
+      throw new Error('A backup ID is required to create a backup.');
+    }
+
+    return new Backup(this, backupId, databasePath, expireTime);
+  }
+
+  getBackups(query?: GetBackupsRequest): Promise<GetBackupsResponse>;
+  getBackups(callback: GetBackupsCallback): void;
+  getBackups(query: GetBackupsRequest, callback: GetBackupsCallback): void;
+  /**
+   * Query object for listing backups.
+   *
+   * @typedef {object} GetBackupsRequest
+   * @property {boolean} [autoPaginate=true] Have pagination handled
+   *     automatically.
+   * @property {number} [maxApiCalls] Maximum number of API calls to make.
+   * @property {number} [maxResults] Maximum number of items to return.
+   * @property {number} [pageSize] Maximum number of results per page.
+   * @property {string} [pageToken] A previously-returned page token
+   *     representing part of the larger set of results to view.
+   */
+  /**
+   * @typedef {array} GetBackupsResponse
+   * @property {Backup[]} 0 Array of {@link Backup} instances.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * List backups on the instance.
+   *
+   * Both completed and in-progress backups are listed if no filter is supplied.
+   *
+   * @see {@link #backup}
+   *
+   * @param query query object for listing backups.
+   * @returns {Promise<GetBackupsResponse>} when resolved, contains a paged list of backups.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const [backups] = await instance.getBackups();
+   */
+  getBackups(
+    queryOrCallback?: GetBackupsRequest | GetBackupsCallback,
+    cb?: GetBackupsCallback
+  ): void | Promise<GetBackupsResponse> {
+    const callback =
+      typeof queryOrCallback === 'function' ? queryOrCallback : cb!;
+    const query =
+      typeof queryOrCallback === 'object'
+        ? queryOrCallback
+        : ({} as GetBackupsRequest);
+
+    const reqOpts = extend({}, query, {
+      parent: this.formattedName_,
+    });
+    this.request<IBackup[]>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'listBackups',
+        reqOpts,
+        gaxOpts: query,
+      },
+      (err, rowBackups, ...args) => {
+        let backups: Backup[] | null = null;
+        if (rowBackups) {
+          backups = rowBackups.map(rowBackup => {
+            const rowBackupName = rowBackup.name!;
+            const backupId = rowBackupName.substring(
```

Comment:
Would you mind adding a comment in the code explaining what this is doing?

Reply:
Oh this is way more complicated than is necessary. I made Backup accept an id or a name so now we can just pass in the name and not have to do this id extraction step.